### PR TITLE
should fix #1726

### LINKED ISSUE #1726: review code for organism.commonName not being unique
commonName is not unique (unlike the ID), so we need to review the code to make sure that we use ID where we can

### PR DESCRIPTION
**Pull request by nathandunn**
fixes #1726

### CODE REVIEW

**Comment by nathandunn** [comment]
@deepakunni3  Ready for review.

File: grails-app/controllers/org/bbop/apollo/OrganismController.groovy [hunk 1]
```diff
@@ -44,11 +44,8 @@ class OrganismController {
             log.debug "deleteOrganism ${organismJson}"
             if (permissionService.isUserAdmin(permissionService.getCurrentUser(organismJson))) {
 
-                log.debug "organism ID: ${organismJson.id}"
-                Organism organism = Organism.findById(organismJson.id as Long)
-                if (!organism) {
-                    organism = Organism.findByCommonName(organismJson.organism)
-                }
+                log.debug "organism ID: ${organismJson.id} vs ${organismJson.organism}"
+                Organism organism = Organism.findById(organismJson.id as Long) ?: Organism.findByCommonName(organismJson.organism)
```
Comment (deepakunni3) [inline grails-app/controllers/org/bbop/apollo/OrganismController.groovy]: Why does it not use `preferenceService.getOrganismForTokenInDB()` here?

Comment (nathandunn) [inline grails-app/controllers/org/bbop/apollo/OrganismController.groovy]: I thought about using it here, but in one case it uses organismJson.id and in the other it uses organismJson.organism so I can't pass the same token in.    But I like this change.

Comment (deepakunni3) [inline grails-app/controllers/org/bbop/apollo/OrganismController.groovy]: Ah, okay

**Comment by deepakunni3** [comment]
@nathandunn Looks good 👍  Did not run into any issues while testing.

**Pull request merged**